kyo-core/shared/src/main/scala/kyo/sums.scala

```diff
@@ -1,25 +1,33 @@
 package kyo
 
 import kyo.core.*
-
-object Sums:
-    private object sums extends Sums[Any]
-    def apply[V]: Sums[V] = sums.asInstanceOf[Sums[V]]
+import scala.annotation.implicitNotFound
+import scala.util.NotGiven
 
 class Sums[V] extends Effect[Sums[V]]:
-    opaque type Command[T] = V
-
-    def add(v: V)(using Tag[Sums[V]]): Unit < Sums[V] =
-        this.suspend[Unit](v)
-
-    def run[T: Flat, S](v: T < (Sums[V] & S))(
-        using Tag[Sums[V]]
-    ): (Chunk[V], T) < S =
-        this.handle(handler)(Chunks.init, v)
+    type Command[T] = V
 
     private val handler =
         new ResultHandler[Chunk[V], Const[V], Sums[V], [T] =>> (Chunk[V], T), Any]:
             def done[T](st: Chunk[V], v: T) = (st, v)
             def resume[T, U: Flat, S](st: Chunk[V], command: V, k: T => U < (Sums[V] & S)) =
                 Resume(st.append(command), k(().asInstanceOf[T]))
 end Sums
+
+object Sums:
+    private object sums extends Sums[Any]
+    private def sums[V]: Sums[V] = sums.asInstanceOf[Sums[V]]
+
+    def add[V](v: V)(using Tag[Sums[V]]): Unit < Sums[V] =
+        sums[V].suspend[Unit](v)
```

Comment:
Just to be clear, though this is a polymorphic call site, the handler should only be allocated once, right?

Does this have any performance benefit?
```
private def sumsHandler[V]: Sums[V] = sums.handler.asInstanceOf[..]
```

Reply:
ah I realize now that you are not invoking the handler, but the method on the effect. It's fine as is.